src/rgw/rgw_op.cc

```diff
@@ -110,23 +112,47 @@ static int parse_range(const char *range, off_t& ofs, off_t& end, bool *partial_
     goto done;
 
   *partial_content = true;
+  end = -2;
```

Comment:
this is for a range that's "x-" -- so end has to be returned as -1 to say "to the end of the file".  the later end tests already know "-1" is an indefinite "end of the file" value.